subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationResult.java

```diff
@@ -151,6 +194,50 @@ public File getOutputLocation() {
             public File resolveForInputArtifact(File inputArtifact) {
                 return outputFile;
             }
+
+            @Override
+            public void visitResult(TransformationResultVisitor visitor) {
+                visitor.visitOutput(outputFile);
+            }
+        }
+    }
+
+    interface TransformationResultVisitor {
+        void visitInput();
+        void visitInput(String relativePath);
+        void visitOutput(File outputLocation);
+    }
```

Comment:
Let's add some documentation for these methods. Why does one `visitInput()` take no parameters, and what is the `relativePath` relative to? The difference between visiting the entire input artifact or part of it should be reflected in the method names, too.

Reply:
I added some Javadoc and renamed the methods.